src/services/cooperation.js

```diff
@@ -91,7 +94,11 @@ const cooperationService = {
       await Cooperation.findByIdAndUpdate(id, { price, needAction: updatedNeedAction }).exec()
     }
     if (status) {
-      await Cooperation.findByIdAndUpdate(id, { status }).exec()
+      const isRequestToClose = status === COOPERATION_STATUS_ENUM[4]
+      const otherRole = currentUserRole === roles.STUDENT ? roles.TUTOR : roles.STUDENT
+      const updatedNeedAction = isRequestToClose ? otherRole : undefined
```

Comment:
I’m not sure about using a ternary operator and explicitly assigning undefined here. It’s especially confusing in this case, as it might give the impression that needAction will be set to undefined in the database (for all statuses except "request to close")

Perhaps we could write it like this instead:

```
const updatedData = { status };

if (isRequestToClose) {
    updatedData.needAction = otherRole;
}

await Cooperation.findByIdAndUpdate(id, updatedData, { runValidators: true });

```